### PR TITLE
Threading Protection

### PR DESCRIPTION
**Pull request by bennowotny**
Provides internal lock mechanisms for `Motor`s and `Group`s so ROS threads don't cause unwanted data races.

This should wait/sync merge with #2.

### CODE REVIEW

File: src/group.cpp [hunk 1]
```diff
@@ -1,22 +1,32 @@
 #include "group.h"
+#include <mutex>
```
Comment (Tzanccc) [inline src/group.cpp]: `group.h` already includes `mutex`, should be removed

File: src/motor.h [hunk 2]
```diff
@@ -29,6 +30,7 @@ namespace MotorLib {
             void resetSettings() const;
 
         private:
+            mutable std::mutex m;
```
Comment (Tzanccc) [inline src/motor.h]: Inconsistent naming of the `mutex` object with `group.h`

**Review by Tzanccc — CHANGES_REQUESTED** [review]
Minor changes required, logged in comments.

File: src/group.cpp (hunk 1 above)
Comment (bennowotny) [reply src/group.cpp]: Fixed by accident during sync merge (a28765a).

File: src/motor.h (hunk 2 above)
Comment (bennowotny) [reply src/motor.h]: Fixed in bac9e62.

File: src/hardware/src/Motor.cpp [hunk 3]
```diff
@@ -9,18 +9,23 @@ namespace Hardware{
         deviceID = ID;
     }
     void Motor::setPower(double power) {
+        const std::lock_guard lock{this->mutex};
```
Comment (Tzanccc) [inline src/hardware/src/Motor.cpp]: Didn't catch this in the first review but `Motor.cpp` uses `this->mutex` while `Group.cpp` uses `mutex`. Might consider changing for consistency

**Review by Tzanccc — CHANGES_REQUESTED** [review]

File: src/hardware/src/Motor.cpp (hunk 3 above)
Comment (bennowotny) [reply src/hardware/src/Motor.cpp]: Fixed in 75bf686.

**Comment by bennowotny** [comment]
@Tzanccc If you like how the other comments are address, please mark them as Resolved.

**Review by Tzanccc — APPROVED** [review]

**Pull request merged**